assess/generators/event_generator.py

```diff
@@ -1,6 +1,19 @@
+from assess.events.events import ProcessStartEvent, ProcessExitEvent, TrafficEvent
+
+
 class EventGenerator(object):
-    def __init__(self, streamer=None):
+    def __init__(self, streamer=None, supported=None):
         self._streamer = streamer
+        if supported is None:
+            if hasattr(streamer, "supported"):
```

Comment:
You can use ``getattr`` for get-or-default in one go
```suggestion
            supported = getattr(streamer, "supported", {ProcessStartEvent: True, ...}):
```

Reply:
Thanks, I included this.